### ISSUE TITLE
Workflows don't fail when calculation fails (SQLA backend)

### ISSUE THREAD

**Issue by szoupanos** [aiidateam/aiida_core#895]
E.g. 

```
+ Workflow PwWorkflow (pk: 44) is Choice(code=RUNNING, value=RUNNING) [4D:18h:35m ago]
|-* Step: start [->run_pw] is Choice(code=FINISHED, value=FINISHED)
|
|-* Step: run_pw [->run_pw] is Choice(code=RUNNING, value=RUNNING)
|
| + Workflow PwrestartWorkflow (pk: 92) is Choice(code=RUNNING, value=RUNNING) [4D:18h:35m ago]
| |-* Step: start [->run_pw_restart] is Choice(code=FINISHED, value=FINISHED)
| | | Calculation (pk: 976) is FINISHED
| |
| |-* Step: run_pw_restart [->run_pw_restart] is Choice(code=ERROR, value=ERROR)
| | | Calculation (pk: 1196) is FAILED
| | | Calculation (pk: 1551) is FAILED
| | | Calculation (pk: 2346) is FAILED
| | | Calculation (pk: 2683) is FAILED
| | | Calculation (pk: 3085) is FAILED
| |
|
```

The problem occurred with the same scripts that were mentioned in  #883

**Comment by szoupanos** [comment]
If you need to release 0.10.0 shift this to 0.10.1
I would like to perform a few more tests before merging.

**Comment by szoupanos** [comment]
This seems to be OK and I think that can be merged and closed.

```
(aiidapy)aiida_test@theospc22:~/wftest$ verdi workflow list  -a
11/09/2017 01:02:59 PM, alembic.runtime.migration: [INFO] Context impl PostgresqlImpl.
11/09/2017 01:02:59 PM, alembic.runtime.migration: [INFO] Will assume transactional DDL.

+ Workflow PwWorkflow (pk: 1) is Choice(code=ERROR, value=ERROR) [1h:38m:31s ago]
|-* Step: start [->run_pw] is Choice(code=FINISHED, value=FINISHED)
|
|-* Step: run_pw [->run_pw] is Choice(code=ERROR, value=ERROR)
|
| + Workflow PwrestartWorkflow (pk: 2) is Choice(code=ERROR, value=ERROR) [1h:37m:59s ago]
| |-* Step: start [->run_pw_restart] is Choice(code=FINISHED, value=FINISHED)
| | | Calculation (pk: 718) is FINISHED
| |
| |-* Step: run_pw_restart [->run_pw_restart] is Choice(code=ERROR, value=ERROR)
| | | Calculation (pk: 727) is FAILED
| |
|
```

**Comment by sphuber** [comment]
Fixed in PR #898
